src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/redundantmodifier/InputRedundantModifierFinalUnnamedVariablesWithOldVersion.java

```diff
@@ -0,0 +1,41 @@
+/*
+RedundantModifier
+tokens = (default)METHOD_DEF, VARIABLE_DEF, ANNOTATION_FIELD_DEF, INTERFACE_DEF, \
+         CTOR_DEF, CLASS_DEF, ENUM_DEF, RESOURCE
+jdkVersion = 8
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
```

Comment:
This is an akward test. It is only compilable by 21, but we set the Java version to 8.

I would split this into 2 tests. One which is actually like Java 8 (I understand we still compile with 11), and the other which is 21. The 21 tests needs a reason it is 21 while we set the Java version to something else.

I am sure we did this with the other PR. If you want to create a separate PR fix, it would be appreciated.

Reply:
> One which is actually like Java 8 (I understand we still compile with 11),

variables named `_` compiles only on Java 8 and below and on Java 21 and below. So we have no other option to used java21 in files that have `_` in them.

Reply:
ok.